src/lib/concurrency/transaction_context.hpp

```diff
@@ -15,46 +15,55 @@ class CommitContext;
- *      |                             |
- *   Rollback operators             Commit operators
- *      |                             |
- *  +-----------------+          Wait for all previous
- *  | ErrorRolledBack |        transaction to be committed
- *  +-----------------+               |
- *                                 +-----------+
- *                                 | Committed |
- *                                 +-----------+
+ *       +--------+
+ *       | Active |
+ *       +--------+
+ *           |
+ *   Execute operators ------------------------+
+ *           |                                 |
+ *           | IF (an operator failed)         | ELSE
+ *           |                                 |
+ *           |                          Finish Transaction -----------------------+
+ *           |                                 |                                  |
+ *           |                                 | IF (User requests Commit         | ELSE (User requests Rollback)
+ *       +---------+                           |     or Auto-Commit-Mode)         |
+ *       | Aborted |                     +------------+                   +------------------+
+ *       +---------+                     | Committing |                   | RolledBackByUser |
+ *           |                           +------------+                   +------------------+
+ *   Rollback operators                        |
+ *           |                           Commit operators
+ *  +-------------------------+                |
+ *  | RolledBackAfterConflict |       Wait for all previous
+ *  +-------------------------+    transaction to be committed
+ *                                             |
+ *                                       +-----------+
+ *                                       | Committed |
+ *                                       +-----------+
+ *
+ *  RolledBackAfterConflict and RolledBackByUser have to be two different transaction phases, because a final transaction
+ *  state of RolledBackAfterConflict is considered as a failure, while RolledBackByUser is considered as a successful
+ *  transaction. Among other things this has an influence on the result message, the database client receives.
  */
 enum class TransactionPhase {
-  Active,                // Transaction has just been created. Operators may be executed.
-  Aborted,               // One of the operators failed. Transaction needs to be rolled back.
-  ErrorRolledBack,       // Transaction has been rolled back due to error.
-  ExplicitlyRolledBack,  // Transaction has been rolled back due to rollback transaction statement.
-  Committing,            // Commit ID has been assigned. Operators may commit records.
-  Committed,             // Transaction has been committed.
+  Active,                       // Transaction has just been created. Operators may be executed.
+  Aborted,                      // One of the operators failed. Transaction needs to be rolled back.
```

Comment:
Old text, but we should change the terminology away from "failure" here, too. Maybe "ran into a transaction conflict"? Next lines, too.

Reply:
@mrks "Failed" is one operator state `enum class ReadWriteOperatorState`.
Should we keep "failed operators" but use "conflicting transactions"?